scripts/build

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+# execute all of the steps necessary to build the project
+set -e #-x
+
+usage() {
+  cat <<-'EOF'
+usage: build [-t|--skip-tests] [-h|--help] BUILDTYPE
+
+[-t|--skip-tests]       skip running unit tests;
+                        this is used in the CI process
+[-h|--help]             print this help text
+
+BUILDTYPE is one of the following:
+- project               the default build command
+- docker                build a docker image
```

Comment:
I still vote for "lint" being included.

Reply:
... I'll look again, but I didn't see a "lint" step explicitly called out in the build anywhere. Would you happen to know where one might exist?

Reply:
I wasn't looking in this repo, just thinking of other repos (https://github.com/DataBiosphere/terra-landing-zone-service/pull/421/files#diff-1641870daa39645825488c81f1ba274a062d49f96635e19297b80f05a21582ad).

Reply:
... would you recommend it being added as a no-op?

or can/should we add it when this repo has some sort of static linting / formatting / something added?

Reply:
Hi, sorry I haven't been around for the code reviews, I do have an opinion about this though.

I think the spotless check should be run as part of `build`. I don't think it's helpful to have a separate `lint` step. Although maybe that's where the `sonar` and `srcclr` steps could go?

Reply:
this is configured here for this project:
`buildSrc/src/main/groovy/bio.terra.java-common-conventions.gradle`